### PR TITLE
Gitops ztp spokes

### PR DESCRIPTION
**Pull request by nsilla**
build-depends: 31585
build-depends: https://github.com/dci-labs/bos2-ci-config/pull/139
TestBOS2: -p sno_baremetal ztp-spoke openshift-ztp-spoke:ansible_extravars=hub_kubeconfig_path:/var/lib/dci-openshift-agent/clusterconfigs-server05/kubeconfig

### CODE REVIEW

**Comment by nsilla** [comment]
recheck

**Review by ramperher — COMMENTED** [review]
After reviewing all the dependencies, and comparing this to the manual deployment I made in Bevo lab, writing some comments here.

File: galaxy.yml
Comment (nsilla) [reply galaxy.yml:105]: Since in the long run we're expected to move to kubernetes.core, I thought of start using the package in the new modules we created.

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml:34]: Done.

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml:1]: good point.

Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml:1]: done

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml:3]: good point. Will do

File: roles/install_operator_gitops/tasks/main.yaml
Comment (nsilla) [reply roles/install_operator_gitops/tasks/main.yaml:43]: done

File: roles/configure_ztp_gitops_apps/defaults/main.yaml
Comment (nsilla) [reply roles/configure_ztp_gitops_apps/defaults/main.yaml:3]: sounds reasonable.
Done.

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nocturnalastro) [inline roles/gitops_configure_repo/tasks/main.yaml:20]: I feel like slurp should be used rather than cat here but that's personal preference.

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nocturnalastro) [inline roles/gitops_configure_repo/tasks/main.yaml:3]: gitlab in title. 

Also I think these are only needed if you have a private repo. So perhaps these should be in a block with some check that indicates that the repo is private.

File: roles/gitops_configure_repo/README.md
Comment (nocturnalastro) [inline roles/gitops_configure_repo/README.md:11]: Don't think we need the reference to gitlab here.

**Review by nocturnalastro — CHANGES_REQUESTED** [review]
I think the role name `gitops_configure_repo` could be improved  to denote its only needed for private repos, perhaps something like `gitops_configure_private_repo`

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (reihl) [reply roles/gitops_configure_repo/tasks/main.yaml:20]: Even better, using 
```
ansible.builtin.set_fact:
  ssh_key:"{{ lookup('file', ssh_key_path) }}"
```
and it save a task

Comment (nocturnalastro) [reply roles/gitops_configure_repo/tasks/main.yaml:20]: If its on the runner yes :)

File: roles/configure_ztp_gitops_apps/tasks/main.yaml
Comment (nsilla) [reply roles/configure_ztp_gitops_apps/tasks/main.yaml:1]: done

File: roles/gitops_configure_repo/README.md
Comment (nsilla) [reply roles/gitops_configure_repo/README.md:11]: Thanks for the comment. I removed the reference to gitlab.

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml:3]: For now I only removed the reference to gitlab.

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml]: Thanks for the comments.
I updated the way we consume the ssh key.
Regarding the name of the role, I was planning on keeping it aligned with the origin role in the eco-validation repository, So for now I'd rather not rename it.

**Comment by fredericlepied** [comment]
recheck

File: galaxy.yml
Comment (manurodriguez) [reply galaxy.yml:106]: @tonyskapunk that has been working on the transition to kubernetes.core in https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/214 , Tony, do you foresee any issue by doing in parts?

File: roles/configure_ztp_gitops_apps/tasks/main.yaml
Comment (ramperher) [inline roles/configure_ztp_gitops_apps/tasks/main.yaml:27]: The `ocp_pull_secret` is a variable that is required for this role, this should be placed in the README + prefix it with `ztp_`

File: roles/configure_ztp_gitops_apps/tasks/main.yaml
Comment (ramperher) [inline roles/configure_ztp_gitops_apps/tasks/main.yaml:142]: The name of this task should be a bit different, something like "Patch the ArgoCD instance to install the GitOps ZTP plugin", just to follow what the docs says: https://docs.openshift.com/container-platform/4.15/edge_computing/ztp-preparing-the-hub-cluster.html#ztp-configuring-hub-cluster-with-argocd_ztp-preparing-the-hub-cluster

File: roles/configure_ztp_gitops_apps/tasks/main.yaml
Comment (ramperher) [inline roles/configure_ztp_gitops_apps/tasks/main.yaml:148]: Where is `ztphub_kubeconfig_file` defined? Shouldn't you use `"{{ temp_dir.path }}/kubeconfig"` instead, similarly to what's done in the patch of the ArgoCD instance (the previous task)?

**Review by ramperher — COMMENTED** [review]
Good to see this green! Here you can find some suggestions to improve the structure of the roles and the documentation. Should be easy to address all them.

File: roles/install_operator_gitops/tasks/main.yaml
Comment (nsilla) [reply roles/install_operator_gitops/tasks/main.yaml:1]: moved back to community.kubernetes.

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml:1]: done

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml:21]: done.

File: roles/gitops_configure_repo/README.md
Comment (nsilla) [reply roles/gitops_configure_repo/README.md:18]: Actually, the variable is not mandatory based on a request from @nocturnalastro. So I updated the configuration.

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml:16]: The variable should be kept optional.

**Comment by ramperher** [comment]
@nocturnalastro , pinging you here to confirm if everything is fine regarding what you asked some days ago, since "change requested" flag is still present from your side, thanks!

**Review by nocturnalastro — DISMISSED** [review]

**Comment by nocturnalastro** [comment]
The changes I requested were fixed as far as I can tell :)

File: roles/gitops_configure_repo/tasks/main.yaml
Comment (nsilla) [reply roles/gitops_configure_repo/tasks/main.yaml:15]: Done.

File: roles/gitops_configure_repo/templates/ssh_private_repo.yaml.j2
Comment (nsilla) [reply roles/gitops_configure_repo/templates/ssh_private_repo.yaml.j2:10]: done.

File: roles/configure_ztp_gitops_apps/tasks/main.yaml
Comment (nsilla) [reply roles/configure_ztp_gitops_apps/tasks/main.yaml:27]: Done.

File: roles/configure_ztp_gitops_apps/tasks/main.yaml
Comment (nsilla) [reply roles/configure_ztp_gitops_apps/tasks/main.yaml:142]: Done.

File: roles/configure_ztp_gitops_apps/tasks/main.yaml
Comment (nsilla) [reply roles/configure_ztp_gitops_apps/tasks/main.yaml:148]: This comes from a copy and paste of the task. The kubeconfig variable to be used should be czga_kubeconfig_path.
Updated.

File: roles/configure_ztp_gitops_apps/README.md
Comment (ramperher) [inline roles/configure_ztp_gitops_apps/README.md:29]: default value is `policies-sub`

**Review by ramperher — COMMENTED** [review]
@nsilla , only one super-minor comment to address, rest looks good

File: roles/configure_ztp_gitops_apps/tasks/main.yaml
Comment (ramperher) [inline roles/configure_ztp_gitops_apps/tasks/main.yaml:148]: If you've copied this in `{{ temp_dir.path }}/kubeconfig`, why not using this here too, like in the other commands that are running in this playbook?

**Review by ramperher — COMMENTED** [review]
I was about to say the issue I've seen with the kubeconfig in https://www.distributed-ci.io/jobs/774ebd45-3c95-459f-b5ee-90bfbae62e15/jobStates?sort=date&task=1aeae134-6ee1-4523-b106-bd4c8b4edfca, but now that I've seen your update here, proposing another suggestion.

File: roles/configure_ztp_gitops_apps/README.md
Comment (ramperher) [reply roles/configure_ztp_gitops_apps/README.md:29]: my bad, this is fine

**Review by ramperher — APPROVED** [review]
You have my ACK to go ahead with this, thanks @nsilla for this great work!

**Comment by ramperher** [comment]
@rlmanrique , since you're the only person available from the `ztp` org (@fredericlepied is off this week), could you take a look to check if it's fine to merge this? That's the only step we've pending before merging this, thanks!

**Review by betoredhat — APPROVED** [review]
LGTM

**Comment by rlmanrique** [comment]
LGTM

**Review by rlmanrique — APPROVED** [review]

**Pull request merged**
